app/providers/registry.py

```
@@ -0,0 +1,114 @@
+        RuntimeError indicating that the legacy registry is not supported.
+    """
+    message = (
+        f"app.providers.registry.{function_name}() is deprecated and the "
+        "legacy provider registry is no longer used. Please use the "
+        "public provider access functions exposed by the 'app.providers' "
+        "package instead."
+    )
+    return RuntimeError(message)
+
+
+def register_provider(provider: CatalogProvider) -> None:
```

Ruff (0.14.13)

Unused function argument: `provider`

(ARG001)